example-dotnet/Program.cs

```diff
@@ -11,8 +22,134 @@ static void Main(string[] args)
+            using (Ionic.Zip.ZipFile zip = Ionic.Zip.ZipFile.Read(@"resources" + Path.DirectorySeparatorChar + "example.zip")) {
+                Console.WriteLine("Files found in zip:");
+                foreach (Ionic.Zip.ZipEntry e in zip) {
+                    Console.WriteLine(e.FileName);
+                }
+            }
+
+            // Usage of SharpCompress API
+
+	    // This uses vulnerable zip extraction API of
+	    // SharpCompress 0.17.1. This library is reported to have
+	    // an 'Arbitrary File Write' (CVE-2018-1002206)
+	    // vulnerability. The vulnerability is fixed in 0.21.0. In
+	    // addition, we also use the class
+	    // `ReadOnlyAppendingStream` which does not exist in
+	    // 0.21.0. 0.17.1 includes net35, net45, netstandard1.0,
+	    // and netstandard1.3 framework versions, whereas 0.21.0
+	    // in addition includes netstandard2.0 framework
+	    // version. This case demonstrates a definite breaking
+	    // update from 0.17.1 to 0.21.0 over different sets of
+	    // frameworks between the used version 0.17.1 and the
+	    // recommended update.
+
+            // Unzip the file to current directory using SharpCompress
+            Console.WriteLine("Unzipping example ZIP file using SharpCompress");
+      
+            // https://github.com/adamhathcock/sharpcompress/wiki/API-Examples
+            var archive = ArchiveFactory.Open("resources" + Path.DirectorySeparatorChar + "example.zip");
+            foreach (var entry in archive.Entries) {
+                if (!entry.IsDirectory) {
+                    Console.WriteLine("Extracting: " + entry.Key);
+                    entry.WriteToDirectory(@".", new ExtractionOptions() { ExtractFullPath = true, Overwrite = true });
+                }
+            }
+
+            IEnumerable<Stream> streamsList = new List<Stream>();
+
+            // This is an API that was broken exists in 0.17.1, but
+            // removed in 0.23.0.
+	    ReadOnlyAppendingStream appendingStream =
+	        new ReadOnlyAppendingStream(streamsList);
+
+	    // This is another breaking API usage, the type of the
+	    // second argument is changed to GZipWriterOptions type in
+	    // 0.18.0.
+	    GZipWriter gzipWriter = new GZipWriter(appendingStream, true);
+            gzipWriter.Dispose();
+
+            // Usage of recurly-client-api API.
+
+	    // recurly-api-client 1.8.0 has two vulnerabilities: 'XML
+	    // External Entity (XXE)' and 'CVE-2017-0907: Server-Side
+	    // Request Forgery (SSRF)'. Depending on the platform
+	    // database, the agent may recommend upgrading to 1.11.4
+	    // or 1.8.1 (if it reports only the SSRF), both of these
+	    // and 1.8.0 include only net35 framework version. This is
+	    // to exemplify the negative case of breaking update over
+	    // the same framework version between the current version
+	    // and the recommended update.
+
+            // Here we simply access an enum type.

```

Comment:
Enums also aren't picked up, so this is a no-op.

Reply:
This enum access has been removed.